Add support for using different health check and upstream ports

greenboxal opened the pull request:
This allows the usage of routers that have different ports for upstream and health check ports.

It also makes clear the distinction between the health check port and the upstream one, as this PR started
when we were trying to have different routers in the same machine and we thought that the port 9999 was hardcoded.

The legacy behaviour of using the health check port as the upstream port is maintained for backward compatibility.

greenboxal commented on aws/cf_template.go:
```diff
@@ -119,7 +124,8 @@ func generateTemplate(certs map[string]time.Time, idleConnectionTimeoutSeconds u
 		HealthCheckIntervalSeconds: cloudformation.Ref("TargetGroupHealthCheckIntervalParameter").Integer(),
 		HealthCheckPath:            cloudformation.Ref("TargetGroupHealthCheckPathParameter").String(),
-		Port:                       cloudformation.Ref("TargetGroupHealthCheckPortParameter").Integer(),
+		HealthCheckPort:            cloudformation.Ref("TargetGroupHealthCheckPortParameter").String(),
```
It was complaining about it being an integer. Will check.

mikkeloscar replied:
True, it is a string: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-elasticloadbalancingv2-targetgroup.html#cfn-elasticloadbalancingv2-targetgroup-healthcheckport weird API :)

greenboxal commented on controller.go:
```diff
@@ -69,6 +70,8 @@ func loadSettings() error {
 		"sets the health check path for the created target groups")
 	flag.UintVar(&healthCheckPort, "health-check-port", aws.DefaultHealthCheckPort,
 		"sets the health check port for the created target groups")
+	flag.UintVar(&upstreamPort, "upstream-port", 0,
```
By doing that it will break backward compatibility. Is that ok?

mikkeloscar replied:
I see!

Actually I would rather break backwards compatibility here instead of carrying over this weird assumption that healthcheckport -> port.

@szuecs what do you think about it?

szuecs replied:
I am in for breaking backwards compatibility, but please add a note in the readme similar to the ones we already have to explain the changes and what you have to do to update and might need to clean up.
Thanks for your great work!

greenboxal commented:
Sorry for not doing this earlier, just addressed your comments:

* Added section to README
* Removed backward compatibility and added note to README
  - I didn't add to the top section about upgrades because I don't know which one to use. Are we releasing a new minor/major with this?

mikkeloscar commented:
@greenboxal what do you think about renaming `upstream-port`? (See my initial comment).

mikkeloscar commented:
@greenboxal any update here? I can also take over if you like.

mikkeloscar commented:
@greenboxal I took the liberty to change the name `upstream-port` to `target-port` to get my last comments fixed on this PR.

This is now ready from my PoV.

mikkeloscar commented:
:+1:

szuecs commented:
:+1:
